app/src/main/java/com/example/lastfm/MainActivity.java

```diff
@@ -4,81 +4,78 @@
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
+import android.icu.text.SymbolTable;
 import android.os.Bundle;
-import android.widget.ImageView;
-import android.widget.TextView;
-
-import com.squareup.picasso.Picasso;
-
-import com.example.lastfm.NewsData;
-import com.example.lastfm.NewsAdapter;
-
-
-import java.util.Arrays;
-import java.util.Collection;
+import android.util.Log;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import java.util.ArrayList;
+import java.util.List;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
-    private ImageView newsImage;
-    private TextView textHeader;
-    private TextView textDescription;
-
-
-    private RecyclerView newsRecyclerView;
-    private NewsAdapter newsAdapter;
-
+    private static final String APP_ID = "b4ab3bf82dcb495e182e04cfc1f12b7b";
+    private static final Integer NUMBER_OF_ARTISTS = 40;
+    private static final String PARSER_PARAM = "artists";
+    private static ArtistAdapter artistsAdapter;
+    private static final String TAG = "Retrofit Error tracking";
+    private static List<Artists> artistsDataList = new ArrayList<>();
+    private static Artists artistData = new Artists();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
-
         initRecyclerView();
-
-        loadNews();
+        retrofitRequest();
+    }
+    private static List<Artists> retrofitRequest() {
+        LastFMClient lastFMClient = ServiceGenerator.createService(LastFMClient.class);
+        Call<JsonObject> call = lastFMClient.numberArtists(NUMBER_OF_ARTISTS, APP_ID, "json");
```

Comment:
Why is the third parameter not a constant? Why do u need it?

Reply:
Created a constant. When i do request without third parameter response comes in XML format.